### PR TITLE
DOMA-4109 re fetch tickets on `ticket/index` page

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**

### CODE REVIEW

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/pages/ticket/index.tsx [hunk 1]
```diff
@@ -53,14 +52,87 @@ const TAP_BAR_ROW_GUTTER: [Gutter, Gutter] = [0, 20]
...
+        where: searchTicketsQuery,
+        first: DEFAULT_PAGE_SIZE,
+        skip: (currentPageIndex - 1) * DEFAULT_PAGE_SIZE,
+    }, {
+        fetchPolicy: 'network-only',
+    })
+
+    const [isRefetching, setIsRefetching] = useState(false)
+    useEffect(() => {
+        const handler = setInterval(async () => {
```
Comment (pahaz) [inline apps/condo/pages/ticket/index.tsx:79]: You need to add a reloadable feature flag here!

File: apps/condo/pages/ticket/index.tsx [hunk 2]
```diff
@@ -53,14 +52,87 @@ const TAP_BAR_ROW_GUTTER: [Gutter, Gutter] = [0, 20]
...
+        count: total,
+        objs: tickets,
+        refetch,
+    } = Ticket.useObjects({
+        sortBy,
+        where: searchTicketsQuery,
+        first: DEFAULT_PAGE_SIZE,
+        skip: (currentPageIndex - 1) * DEFAULT_PAGE_SIZE,
+    }, {
+        fetchPolicy: 'network-only',
```
Comment (pahaz) [inline apps/condo/pages/ticket/index.tsx:74]: why the network only?

File: apps/condo/pages/ticket/index.tsx (hunk 1 above)
Comment (nomerdvadcatpyat) [reply apps/condo/pages/ticket/index.tsx]: done

**Review by pahaz — APPROVED** [review]

File: apps/condo/pages/ticket/index.tsx [hunk 3]
```diff
@@ -53,14 +54,96 @@ const TAP_BAR_ROW_GUTTER: [Gutter, Gutter] = [0, 20]
...
+    const [isRefetching, setIsRefetching] = useState(false)
+    useEffect(() => {
+        if (isRefetchTicketsFeatureEnabled) {
+            const handler = setInterval(async () => {
+                setIsRefetching(true)
+                await refetch()
+                setIsRefetching(false)
+            }, TICKETS_RE_FETCH_INTERVAL)
+            return () => {
+                clearInterval(handler)
```
Comment (akarjakin) [inline apps/condo/pages/ticket/index.tsx:95]: will this work if `isRefetchTicketsFeatureEnabled` value changes from `true` to `false`?

File: apps/condo/pages/ticket/index.tsx [hunk 4]
```diff
@@ -53,14 +54,96 @@ const TAP_BAR_ROW_GUTTER: [Gutter, Gutter] = [0, 20]
...
+            }
+        }
+    }, [isRefetchTicketsFeatureEnabled, refetch])
+
+    const tableColumns = useTableColumns(filterMetas, tickets)
+    const loading = isTicketsFetching && !isRefetching
+
+    const handleRowAction = useCallback((record) => {
```
Comment (akarjakin) [inline apps/condo/pages/ticket/index.tsx:103]: this doesn't handle row action, it, instead, returns object with such handler. I think here we have naming issue.

File: apps/condo/pages/ticket/index.tsx [hunk 5]
```diff
@@ -53,14 +54,96 @@ const TAP_BAR_ROW_GUTTER: [Gutter, Gutter] = [0, 20]
...
+    const { useFlag, updateContext } = useFeatureFlags()
+    const { organization } = useOrganization()
+
+    useEffect(() => {
+        updateContext({ organization: organization.id })
+    }, [organization, updateContext])
+
+    const isRefetchTicketsFeatureEnabled = useFlag(RE_FETCH_TICKETS_IN_CONTROL_ROOM)
+
+    const [isRefetching, setIsRefetching] = useState(false)
```
Comment (sitozzz) [inline apps/condo/pages/ticket/index.tsx:86]: We decided to place state definitions at the top of component code)

**Review by sitozzz — COMMENTED** [review]
Looks good

**Review by sitozzz — COMMENTED** [review]
Looks good

**Pull request merged**